src/bin.ts

```diff
@@ -54,16 +54,43 @@ v8flags(function (err, v8flags) {
   const proc = spawn(
     process.execPath,
     nodeArgs.concat(join(__dirname, '_bin.js'), scriptArgs),
-    { stdio: 'inherit' }
+    {
+      // We need to run in detached mode so to avoid
+      // automatic propagation of signals to the child process.
+      // This is necessary because by default, keyboard interrupts
+      // are propagated to the process tree, but `kill` is not.
+      //
+      // See: https://nodejs.org/api/child_process.html#child_process_options_detached
+      detached: true,
+
+      // Pipe all input and output to this process
+      stdio: 'inherit'
+    }
   )
 
-  proc.on('exit', function (code: number, signal: string) {
-    process.on('exit', function () {
-      if (signal) {
-        process.kill(process.pid, signal)
-      } else {
-        process.exit(code)
-      }
-    })
+  // Ignore signals, and instead forward them to the
+  // child process
+  const forward = (signal: string) => process.on(signal, () => proc.kill(signal))
+
+  // Interrupt (CTRL-C)
+  forward('SIGINT')
+
+  // Termination (`kill` default signal)
+  forward('SIGTERM')
+
+  // Terminal size change must be forwarded to the subprocess
+  forward('SIGWINCH')
+
+  // On exit, exit this process with the same exit code
+  proc.on('close', (code: number, signal: string) => {
+    if (signal) {
+      process.kill(process.pid, signal)
+    } else if (code) {
```

Comment:
Do we need to check the `else if` here? If it closes, shouldn't it have one or the other? In a hypothetical case it has neither, wouldn't we still want to exit?

Reply:
I *think* we want the process to exit naturally (no process.exit) upon a signal, but to exit with a code when there is one.